py-polars/polars/series/series.py

```diff
@@ -326,10 +326,14 @@ def __init__(
             )
             if values.dtype.type in [np.datetime64, np.timedelta64]:
                 # cast to appropriate dtype, handling NaT values
+                input_dtype = _resolve_temporal_dtype(None, values.dtype)
+                # numpy dtype has already been validated in `numpy_to_pyseries`
+                assert input_dtype is not None
```

Comment:
Don't think we should leave the `assert` here 🤔

Reply:
I put it for mypy - would you just `# type: ignore` or cast instead?

Reply:
> I put it for mypy - would you just `# type: ignore` or cast instead?

Yup, a `type: ignore[...]` is preferable :sunglasses:👍 
(No runtime cost/behaviour)